Cargo.toml

```diff
@@ -90,6 +90,7 @@ thiserror = "1.0"
 test-log = "0.2"
 fs4 = { version = "0.8.3", features = ["sync"] }
 ariadne = { version = "0.4.1", features = ["auto-color"] }
+git2 = "0.19.0"   
```

Comment:
This causes the binary size of `moon` to increase from 3.5M to 6.4M (tested on macOS), and binary size is one of our optimization goals.

Reply:
Ok, then I try to split a part of git2 to here